src/Workspaces/CoreTest/SolutionTests/SolutionWithSourceGeneratorTests.cs

```diff
@@ -91,6 +91,28 @@ public async Task WithReferencesMethodCorrectlyUpdatesRunningGenerators()
             Assert.Single((await project.GetRequiredCompilationAsync(CancellationToken.None)).SyntaxTrees);
         }
 
+        // We only run this test on Release, as the compiler has asserts that trigger in Debug that the type names probably shouldn't be the same.
```

Comment:
@chsienki This is because of the assert that's on the GeneratorDriver constructor...not sure if we want to change it or not.

Reply:
yeah, tahts' a bad assert.  it should either go or throw.

Reply:
@chsienki let's figure out the plan as a part of https://github.com/dotnet/roslyn/issues/56619, and remember to fix this test with whatever we decide.